src/open_inwoner/openzaak/tests/test_case_detail.py

```diff
@@ -818,6 +818,57 @@ def test_second_status_preview(self, m):
                 res = detail_view.get_second_status_preview(status_types)
                 self.assertEqual(res, result)
 
+    def test_e_suite_statustype_zaaktype_mapping(self, m):
+        """
+        Regression test for #1991
+
+        eSuite has one statustype for multiple zaaktypes, hence it could happen that
+        the wrong statustypeconfig (corresponding to different zaaktype) was used
+        """
+        self.maxDiff = None
+
+        bogus_zaaktype = generate_oas_component(
+            "ztc",
+            "schemas/ZaakType",
+            identificatie="Bogus zaaktype",
+        )
+        bogus_zt_config = ZaakTypeConfigFactory(
+            identificatie=bogus_zaaktype["identificatie"],
+        )
+
+        ZaakTypeStatusTypeConfigFactory(
+            zaaktype_config=self.zaaktype_config,
+            statustype_url=self.status_type_new["url"],
+            status_indicator=StatusIndicators.warning,
+            status_indicator_text="foo",
+            case_link_text="Bekijk aanvraag",
+        )
+        ZaakTypeStatusTypeConfigFactory(
+            zaaktype_config=bogus_zt_config,
+            statustype_url=self.status_type_new["url"],
+            status_indicator=StatusIndicators.warning,
+            status_indicator_text="bogus",
+            case_link_text="bogus",
+        )
+        ZaakTypeStatusTypeConfigFactory(
+            zaaktype_config=self.zaaktype_config,
+            statustype_url=self.status_type_finish["url"],
+            status_indicator=StatusIndicators.success,
+            status_indicator_text="bar",
+            call_to_action_url="https://www.example.com",
+            call_to_action_text="Click me",
+            case_link_text="Bekijk aanvraag",
```

Comment:
It is better for readability to strip the factories and generators from fields that aren't relevant to the test (also a few just above). 

If they are required for the model but not relevant for the test they should be part of the factory.